viewing_party/party.py

```diff
@@ -1,23 +1,175 @@
+# -----------------------------------------
 # ------------- WAVE 1 --------------------
+# -----------------------------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie = {}
+
+    if not title or not genre or not rating:
+        return None
+    else:
+        movie["title"] = title
+        movie["genre"] = genre
+        movie["rating"] = rating
+
+    return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+
+    return user_data   
+
+def watch_movie(user_data, movie):
+    # If the movie is the watchlist, it's moved from the watchlist 
+    # to the watched
+    for i in range(len(user_data["watchlist"])):
+        if movie == user_data["watchlist"][i]["title"]: 
+            user_data["watched"].append(user_data["watchlist"][i])
+            del user_data["watchlist"][i]
+
+    return user_data    
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    if not user_data["watched"]:
+        return 0.0
+    
+    count = 0

```

Comment:
Counter variables can be super useful in many different contexts. The one place we don't necessarily need to use them though is in situations where the counter is modeling data that already exists and is easily accessible. 

In this function, you add one to count for each movie that is in `user_data["watched"]`. This means that count will always just end up being `len(user_data["watched"]`. If we weren't counting every item in the list, I could see the count variable being necessary, but in this case, it's better to go without the new variable and just use direct access of the list's length!